pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -258,6 +263,19 @@ func (c *Reconciler) makeBuild(tr *v1alpha1.TaskRun, logger *zap.SugaredLogger)
 		logger.Errorf("Failed to create tracker for build %s for taskrun %s: %v", buildRef, tr.Name, err)
 		return nil, err
 	}
-
 	return createdBuild, nil
 }
+
+func applyTemplates(b *buildv1alpha1.Build, t *v1alpha1.Task, tr *v1alpha1.TaskRun) *buildv1alpha1.Build {
+	// This assumes that the TaskRun inputs have been validated against what the Task requests.
+	replacements := map[string]string{}
+	for _, p := range tr.Spec.Inputs.Params {
+		replacements[fmt.Sprintf("inputs.params.%s", p.Name)] = p.Value
+	}
+
+	return builder.ApplyReplacements(b, replacements)
+}
```

Comment:
personally id rather leave things in `build` or move them here even - i feel like `pkg` has a lot of random unrelated stuff in it, and i like that the `build` repo has all the `build` stuff in it, it's better factored in my opinion than `pkg`

(#147 might be relevant)

Reply:
But thats the definition of the pkg repo, to include all the common things between knative repos. This is a strong dependency on actual functions in a different repo that are not part of the exposed interface of that repo. I would prefer to even copy it here for the time being.

Reply:
This can be decided later tho, doesn’t have to be part of this PR

Reply:
pkg sounds fine for this if that's the goal of the repo. I'm confused what you mean by exposed surface of the build repo. My understanding was that anything under pkg was fair game. Is there a defined area for exporting packages in that repo?

Reply:
Its fair game to use yes, but its a func to be used by the builder, from an architectural  perspective its a better fit here or in pkg.

Reply:
Its fine either way for now until the future of build is clear

Reply:
> But thats the definition of the pkg repo, to include all the common things between knative repos. 

it seems to me like what is happening is a lot of the stuff in knative is being developed _as if_ it's all in one repo - if this _was_ all in one repo, and we had one directory where we put all our `common` logic, and we called it something like `common` (that seems to me like the alternate name for `pkg`), I'd say that was an example of [package naming that indicates the code could be better factored](https://blog.golang.org/package-names), and I'd push for splitting that up into more focused packages. At the moment I feel like `knative/build` _is_ well factored in that it has all the build related logic in it, and taking the build logic OUT of a well factored location and putting into our pile of `common` logic seems like a step back

i feel very strongly that a better version of the pkg repo would be several well factored repos that contain libs that each do one thing and do it well (e.g. a lot of the logic around building controllers would be really nice in its own repo with docs and examples, so folks outside knative could use it too) - at the moment it seems to be a miscellaneous grab bag and if it's possible to avoid putting more logic in there, id rather do that

sorry for the rant :S

> Its fine either way for now until the future of build is clear

yeah itll probably be more clear what we should do at that that point!

Reply:
agree with all that, but in this case its logic to replace params with values, not really build specific. Its up to you, not a big deal to me, was just stating my opinion

Reply:
I was going to just port the logic over here, but it is actually slightly build specific. If you check out the implementation:
https://github.com/knative/build/blob/92a1258647b272a3a4505616b102d6e8f86be082/pkg/builder/common.go#L60

It applies the string replacement, but to a very specific set of keys/values within build. You could argue that this should probably be "step specific" rather than "build specific", but there is also some logic for BuildTemplates as well.

Are we all good with leaving this as is, then moving it up to pkg at some point once the future of build is clear?

Reply:
I’m good with using this func from Build for now like you have done @dlorenc 
Thanks